src/spi.rs

```diff
@@ -53,20 +53,18 @@ use crate::rcc::{Clocks, Enable, GetBusFreq, Reset, APB1, APB2};
 use crate::time::Hertz;
 
 use core::sync::atomic::{self, Ordering};
-
-use as_slice::AsSlice;
+use embedded_dma::StaticReadBuffer;
 
 /// SPI error
 #[derive(Debug)]
+#[non_exhaustive]
```

Comment:
Are we expecting to add more errors here in the future? IIRC we handle all hardware errors that can occur already.

Reply:
I think it's good practice to avoid breaking changes, but again, it was there before (implemented manually), I just changed to please clippy. I would leave this decision to another PR, I prefer not to change behavior not related to DMA in this one.

Reply:
Oh right, I missed the _Exclusive thing. Yea, let's leave that to the future